R/utilities-mass-balance.R

```diff
@@ -38,3 +38,342 @@ envList2PathStringsVector <- function(envList) {
+  massBalancePlots[["timeProfile"]] <- plotMassBalanceTimeProfile(
+    data = simulationResultsOutputByGroup,
+    metaData = metaDataOutputByGroup,
+    dataMapping = tlf::XYGDataMapping$new(
+      x = "Time",
+      y = "Amount",
+      color = "Legend"
+    ),
+    plotConfiguration = plotConfigurations[["timeProfile"]]
+  )
+
+  massBalancePlots[["cumulativeTimeProfile"]] <- plotMassBalanceCumulativeTimeProfile(
+    data = simulationResultsOutputByGroup,
+    metaData = metaDataOutputByGroup,
+    dataMapping = tlf::XYGDataMapping$new(
+      x = "Time",
+      y = "Amount",
+      fill = "Legend"
+    ),
+    plotConfiguration = plotConfigurations[["cumulativeTimeProfile"]]
+  )
+
+  massBalancePlots[["normalizedTimeProfile"]] <- plotMassBalanceTimeProfile(
+    data = simulationResultsOutputByGroup,
+    metaData = metaDataOutputByGroup,
+    dataMapping = tlf::XYGDataMapping$new(
+      x = "Time",
+      y = "NormalizedAmount",
+      color = "Legend"
+    ),
+    plotConfiguration = plotConfigurations[["normalizedTimeProfile"]]
+  )
+
+  massBalancePlots[["normalizedCumulativeTimeProfile"]] <- plotMassBalanceCumulativeTimeProfile(
+    data = simulationResultsOutputByGroup,
+    metaData = metaDataOutputByGroup,
+    dataMapping = tlf::XYGDataMapping$new(
+      x = "Time",
+      y = "NormalizedAmount",
+      fill = "Legend"
+    ),
+    plotConfiguration = plotConfigurations[["normalizedCumulativeTimeProfile"]]
+  )
+
+  massBalancePlots[["pieChart"]] <- plotMassBalancePieChart(
+    data = simulationResultsOutputByGroup,
+    metaData = metaDataOutputByGroup,
+    dataMapping = tlf::XYGDataMapping$new(
+      x = "Time",
+      y = "NormalizedAmount",
+      fill = "Legend"
+    ),
+    plotConfiguration = plotConfigurations[["pieChart"]]
+  )
+
+  return(list(
+    plots = massBalancePlots,
+    timeProfile = simulationResultsOutputByGroup
+  ))
+}
+
+
+#' @title plotMassBalanceTimeProfile
+#' @description Plot mass balance time profile
+#' @param data data.frame
+#' @param metaData meta data on `data`
+#' @param dataMapping `XYGDataMapping` R6 class object from `tlf` library
+#' @param plotCOnfiguration `PlotConfiguration` R6 class object from `tlf` library
+#' @return ggplot object of time profile for mean model workflow
+#' @export
+#' @import tlf
+#' @import ggplot2
+#' @import utils
+plotMassBalanceTimeProfile <- function(data,
+                                       metaData = NULL,
+                                       dataMapping = NULL,
+                                       plotConfiguration = NULL) {
+  timeVsAmountDataMapping <- dataMapping %||% tlf::XYGDataMapping$new(
+    x = "Time",
+    y = "Amount",
+    color = "Legend"
+  )
+
+  # TO DO: remove Testing Data Watermark when plot is complete
+  plotConfiguration <- plotConfiguration %||% tlf::PlotConfiguration$new(
+    data = data,
+    metaData = metaData,
+    dataMapping = timeVsAmountDataMapping,
+    watermark = "Testing !"

```

Comment:
Should not the watermark be part of the config of this workflow run? E.g. a settings that says this is STILL not QCed or something?

Reply:
Yes, for this I am going to define a default theme for the Reporting Engine.